MonoGame.Framework/Graphics/GraphicsResource.cs

```diff
@@ -115,9 +116,19 @@ internal static void DisposeAll()
                 {
                     var target = resource.Target;
                     if (target != null)
+                    {
+                        // Do not dispose another device's graphics resource
+                        if (target is GraphicsResource)
+                        {
+                            if ((target as GraphicsResource).GraphicsDevice != GraphicsDeviceContext.Current.GraphicsDevice)
+                                continue;
+                        }
                         (target as IDisposable).Dispose();
+                    }
+
+                    // We can remove weak handle from collection safely since we have cloned resources dictionary
+                    resources.Remove(resource);
```

Comment:
@tgjones  - This is another case of a `static` that is graphics device specific.  If you get some time maybe you can move this one into `GraphicsDevice` too?